core/src/main/java/io/grpc/transport/MessageFramer.java

```diff
@@ -181,7 +184,8 @@ private void writeRaw(byte[] b, int off, int len) {
         commitToSink(false);
       }
       if (buffer == null) {
-        buffer = bufferAllocator.allocate(maxFrameSize);
+        // Allocate as big a buffer as the allocator will allow
```

Comment:
Is this comment correct?  Shouldn't it be something like "Allocate a buffer that is at least big enough to for this write"?